test/e2e/external_gateways.go

```diff
@@ -374,6 +377,108 @@ var _ = ginkgo.Describe("e2e multiple external gateway validation", func() {
+		annotateNamespaceForGateway(f.Namespace.Name, false, addresses.gatewayIPs[0], addresses.gatewayIPs[1])
+
+		// ensure the conntrack deletion tracker annotation is updated
+		ginkgo.By("Check if the k8s.ovn.org/external-gw-pod-ips got updated for the app namespace")
+		err := wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			ns := getNamespace(f, f.Namespace.Name)
+			return (ns.Annotations[externalGatewayPodIPsAnnotation] == fmt.Sprintf("%s,%s", addresses.gatewayIPs[0], addresses.gatewayIPs[1])), nil
+		})
+		framework.ExpectNoError(err, "Check if the k8s.ovn.org/external-gw-pod-ips got updated, failed: %v", err)
+
+		setupIperf3Client := func(container, address string, port int) {
+			// note iperf3 even when using udp also spawns tcp connection first; so we indirectly also have the tcp connection when using "-u" flag
+			cmd := []string{"docker", "exec", container, "iperf3", "-u", "-c", address, "-p", fmt.Sprintf("%d", port), "-b", "1M", "-i", "1", "-t", "3", "&"}
+			_, err := runCommand(cmd...)
+			framework.ExpectNoError(err, "failed to setup iperf3 client for %s", container)
+		}
+		macAddressGW := make([]string, 2)
+		for i, containerName := range []string{gwContainer1, gwContainer2} {
+			ginkgo.By("Start iperf3 client from external container to connect to iperf3 server running at the src pod")
+			setupIperf3Client(containerName, addresses.srcPodIP, 5201+i)
+			macAddressExtGW, err := net.ParseMAC(getMACAddressesForNetwork(containerName, externalContainerNetwork))
+			framework.ExpectNoError(err, "failed to parse MAC address for %s", containerName)
+			macAddressGW[i] = strings.Replace(macAddressExtGW.String(), ":", "", -1)
+		}
+
+		ginkgo.By("Check if conntrack entries for ECMP routes are created for the 2 external gateways")
+		nodeName := getPod(f, srcPodName).Spec.NodeName
+		res := pokeConntrackEntries(nodeName, addresses.srcPodIP, protocol, macAddressGW)
+		gomega.Expect(res).To(gomega.Equal(2))
+
+		ginkgo.By("Remove second external gateway IP from the app namespace annotation")
+		annotateNamespaceForGateway(f.Namespace.Name, false, addresses.gatewayIPs[0])
+
+		// ensure the conntrack deletion tracker annotation is updated
+		ginkgo.By("Check if the k8s.ovn.org/external-gw-pod-ips got updated for the app namespace")
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			ns := getNamespace(f, f.Namespace.Name)
+			return (ns.Annotations[externalGatewayPodIPsAnnotation] == fmt.Sprintf("%s", addresses.gatewayIPs[0])), nil
+		})
+		framework.ExpectNoError(err, "Check if the k8s.ovn.org/external-gw-pod-ips got updated, failed: %v", err)
+
+		ginkgo.By("Check if conntrack entries for ECMP routes are removed for the deleted external gateway if traffic is UDP")
+		res = pokeConntrackEntries(nodeName, addresses.srcPodIP, protocol, macAddressGW)
+		if protocol == "udp" {
+			gomega.Expect(res).To(gomega.Equal(1))
```

Comment:
because we added 2 SPKs (gwContainer1, gwContainer2) for this pod and then we deleted one SPK pod (gwContainer1), so we still have the conntrack entry for the second SPK pod (gwContainer2).

Reply:
added a comment to make this clear.